### PR TITLE
infra: prefer OPENCLAW_IMAGE_APT_PACKAGES over OPENCLAW_DOCKER_APT_PACKAGES

### PR DESCRIPTION
**Pull request by urtabajev**
## Summary

- Problem: `OPENCLAW_DOCKER_APT_PACKAGES` is Docker-specific naming for a variable that applies equally to Docker and Podman image builds.
- Why it matters: UX - a generic, container-runtime-agnostic name is clearer for users on any runtime.
- What changed: Introduced `OPENCLAW_IMAGE_APT_PACKAGES` as the preferred variable in the Dockerfile, both setup scripts (Docker and Podman), and all related docs. The legacy `OPENCLAW_DOCKER_APT_PACKAGES` is still accepted as a fallback for backward compatibility.
- What did NOT change (scope boundary): No runtime behavior change. No other env vars renamed. No breaking change - existing users of `OPENCLAW_DOCKER_APT_PACKAGES` are unaffected.

## Change Type (select all)

- [ ] Bug fix
- [x] Feature
- [ ] Refactor required for the fix
- [x] Docs
- [ ] Security hardening
- [x] Chore/infra

## Scope (select all touched areas)

- [ ] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [ ] API / contracts
- [ ] UI / DX
- [x] CI/CD / infra

## Linked Issue/PR

- N/A
- [ ] This PR fixes a bug or regression

## Root Cause (if applicable)

N/A

## Regression Test Plan (if applicable)

N/A - this is an additive env var alias with no logic change beyond variable resolution order.

## User-visible / Behavior Changes

- New preferred env var: `OPENCLAW_IMAGE_APT_PACKAGES` (runtime-agnostic).
- Legacy `OPENCLAW_DOCKER_APT_PACKAGES` continues to work as a fallback.
- If both are set, `OPENCLAW_IMAGE_APT_PACKAGES` takes precedence.

## Diagram (if applicable)

N/A

## Security Impact (required)

- New permissions/capabilities? No
- Secrets/tokens handling changed? No
- New/changed network calls? No
- Command/tool execution surface changed? No
- Data access scope changed? No

### Environment

- OS: Windows 11 / Podman machine (WSL2, debian:bookworm-slim)
- Runtime/container: Podman

### Steps

1. `export OPENCLAW_IMAGE_APT_PACKAGES="curl wget"`
2. Run `./scripts/docker/setup.sh` or `./scripts/podman/setup.sh`
3. Verify the image build passes `--build-arg OPENCLAW_IMAGE_APT_PACKAGES="curl wget"` to the Dockerfile

### Expected

- Packages are installed in the built image via the new variable name.

### Actual

- Same as expected.

### Real Behavior Proof

All four edge cases verified with a live `podman build` using the exact `ARG` / `RUN` block from this PR's `Dockerfile`:

```
ARG OPENCLAW_IMAGE_APT_PACKAGES
ARG OPENCLAW_DOCKER_APT_PACKAGES=""
RUN packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; \
    echo "RESOLVED: packages=${packages}"
```

**Test 1 — new var only:**
```
$ podman build --no-cache -f Dockerfile.proof \
    --build-arg OPENCLAW_IMAGE_APT_PACKAGES="curl wget httpie" .
STEP 4/4: RUN packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; echo "RESOLVED: packages=${packages}"
RESOLVED: packages=curl wget httpie
```

**Test 2 — legacy var only (fallback):**
```
$ podman build --no-cache -f Dockerfile.proof \
    --build-arg OPENCLAW_DOCKER_APT_PACKAGES="curl wget" .
STEP 4/4: RUN packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; echo "RESOLVED: packages=${packages}"
RESOLVED: packages=curl wget
```

**Test 3 — new var wins over legacy:**
```
$ podman build --no-cache -f Dockerfile.proof \
    --build-arg OPENCLAW_IMAGE_APT_PACKAGES="curl wget httpie" \
    --build-arg OPENCLAW_DOCKER_APT_PACKAGES="curl wget" .
STEP 4/4: RUN packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; echo "RESOLVED: packages=${packages}"
RESOLVED: packages=curl wget httpie
```

**Test 4 — explicit empty new var suppresses legacy:**
```
$ podman build --no-cache -f Dockerfile.proof \
    --build-arg OPENCLAW_IMAGE_APT_PACKAGES="" \
    --build-arg OPENCLAW_DOCKER_APT_PACKAGES="curl wget" .
STEP 4/4: RUN packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; echo "RESOLVED: packages=${packages}"
RESOLVED: packages=
```

Shell-level variable normalization (from `scripts/docker/setup.sh` and `scripts/podman/setup.sh`) also verified with live bash:

```
=== Test 1: OPENCLAW_IMAGE_APT_PACKAGES takes precedence ===
Resolved: OPENCLAW_IMAGE_APT_PACKAGES=curl wget httpie

=== Test 2: Legacy OPENCLAW_DOCKER_APT_PACKAGES used as fallback ===
Resolved: OPENCLAW_IMAGE_APT_PACKAGES=curl wget

=== Test 3: Explicit empty OPENCLAW_IMAGE_APT_PACKAGES suppresses legacy ===
Resolved: OPENCLAW_IMAGE_APT_PACKAGES=
```

## Human Verification (required)

- Verified scenarios: All four edge cases (new only, legacy only, both set, empty-new suppresses legacy) confirmed with live `podman build` output and live bash output.
- Edge cases checked: Both vars set (new wins), only legacy set (fallback works), neither set (no-op), explicit empty new var (suppresses legacy).
- What you did **not** verify: Full OpenClaw image build end-to-end (packages actually installed inside the image).

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [ ] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? Yes
- Config/env changes? Yes - new preferred variable `OPENCLAW_IMAGE_APT_PACKAGES`; legacy `OPENCLAW_DOCKER_APT_PACKAGES` still works.
- Migration needed? No

## Risks and Mitigations

- Risk: Users who set both variables may not realize `OPENCLAW_IMAGE_APT_PACKAGES` takes precedence.
  - Mitigation: Documented in Dockerfile comments and docs.

### CODE REVIEW

File: scripts/podman/setup.sh [hunk 1]
```diff
@@ -358,9 +358,10 @@ install -d -m 700 "$OPENCLAW_CONFIG_DIR" "$OPENCLAW_WORKSPACE_DIR"
 ensure_private_existing_dir_owned_by_user "config directory" "$OPENCLAW_CONFIG_DIR"
 ensure_private_existing_dir_owned_by_user "workspace directory" "$OPENCLAW_WORKSPACE_DIR"
 
+OPENCLAW_IMAGE_APT_PACKAGES="${OPENCLAW_IMAGE_APT_PACKAGES:-${OPENCLAW_DOCKER_APT_PACKAGES:-}}"
```
[the remark this answers is not in the document]
Comment (urtabajev) [reply scripts/podman/setup.sh]: Fixed in 9de4d48.

File: Dockerfile [hunk 2]
```diff
@@ -190,13 +190,16 @@ RUN install -d -m 0755 "$COREPACK_HOME" && \
...
 # Install additional system packages needed by your skills or extensions.
-# Example: docker build --build-arg OPENCLAW_DOCKER_APT_PACKAGES="python3 wget" .
+# Example: docker build --build-arg OPENCLAW_IMAGE_APT_PACKAGES="python3 wget" .
+# Legacy alias: OPENCLAW_DOCKER_APT_PACKAGES is still accepted as a fallback.
+ARG OPENCLAW_IMAGE_APT_PACKAGES=""
 ARG OPENCLAW_DOCKER_APT_PACKAGES=""
 RUN --mount=type=cache,id=openclaw-bookworm-apt-cache,target=/var/cache/apt,sharing=locked \
     --mount=type=cache,id=openclaw-bookworm-apt-lists,target=/var/lib/apt,sharing=locked \
-    if [ -n "$OPENCLAW_DOCKER_APT_PACKAGES" ]; then \
+    packages="${OPENCLAW_IMAGE_APT_PACKAGES:-$OPENCLAW_DOCKER_APT_PACKAGES}"; \
```
[the remark this answers is not in the document]
Comment (urtabajev) [reply Dockerfile]: Fixed in 9de4d48.

File: scripts/docker/setup.sh [hunk 3]
```diff
@@ -276,7 +276,7 @@ export OPENCLAW_GATEWAY_PORT="${OPENCLAW_GATEWAY_PORT:-18789}"
 export OPENCLAW_BRIDGE_PORT="${OPENCLAW_BRIDGE_PORT:-18790}"
 export OPENCLAW_GATEWAY_BIND="${OPENCLAW_GATEWAY_BIND:-lan}"
 export OPENCLAW_IMAGE="$IMAGE_NAME"
-export OPENCLAW_DOCKER_APT_PACKAGES="${OPENCLAW_DOCKER_APT_PACKAGES:-}"
+export OPENCLAW_IMAGE_APT_PACKAGES="${OPENCLAW_IMAGE_APT_PACKAGES:-${OPENCLAW_DOCKER_APT_PACKAGES:-}}"
```
[the remark this answers is not in the document]
Comment (urtabajev) [reply scripts/docker/setup.sh]: Fixed in 9de4d48.

File: Dockerfile [hunk 4]
```diff
@@ -190,13 +190,16 @@ RUN install -d -m 0755 "$COREPACK_HOME" && \
...
 # Install additional system packages needed by your skills or extensions.
-# Example: docker build --build-arg OPENCLAW_DOCKER_APT_PACKAGES="python3 wget" .
+# Example: docker build --build-arg OPENCLAW_IMAGE_APT_PACKAGES="python3 wget" .
+# Legacy alias: OPENCLAW_DOCKER_APT_PACKAGES is still accepted as a fallback.
+ARG OPENCLAW_IMAGE_APT_PACKAGES=""
 ARG OPENCLAW_DOCKER_APT_PACKAGES=""
 RUN --mount=type=cache,id=openclaw-bookworm-apt-cache,target=/var/cache/apt,sharing=locked \
     --mount=type=cache,id=openclaw-bookworm-apt-lists,target=/var/lib/apt,sharing=locked \
-    if [ -n "$OPENCLAW_DOCKER_APT_PACKAGES" ]; then \
+    packages="${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES}"; \
```
[the remark this answers is not in the document]
Comment (urtabajev) [reply Dockerfile]: Fixed in 78d43cb.

Root cause: ARG OPENCLAW_IMAGE_APT_PACKAGES="" initializes the variable as set (to empty string) at the Docker layer. The fallback ${OPENCLAW_IMAGE_APT_PACKAGES-$OPENCLAW_DOCKER_APT_PACKAGES} uses - (unset-only), so it never falls back.

Fix: Remove the default so the ARG is bare (ARG OPENCLAW_IMAGE_APT_PACKAGES). Now the variable is truly unset unless --build-arg is passed, and the legacy fallback works.

Regression test added that asserts the ARG line has no default value. Test passes after fix.

**Comment by urtabajev** [comment]
Rebased onto current upstream main (`078e7a65`). CI failures in `check`, `check-additional`, and `checks-fast-extensions-shard-4` are caused by TS/test breakage in `extensions/memory-core/` introduced by recent upstream merges (`feat(memory): add grounded REM backfill lane`, `feat(memory): harden grounded REM extraction`) — these affect all open PRs, not just ours. Our changes only touch `Dockerfile`, `scripts/{docker,podman}/setup.sh`, `docs/install/docker.md`, and `src/docker-setup.e2e.test.ts`. Waiting for upstream fix.

**Comment by steipete** [comment]
Maintainer verification before merge:

Behavior addressed: `OPENCLAW_IMAGE_APT_PACKAGES` is the preferred runtime-neutral image build arg for Docker/Podman apt package installs; `OPENCLAW_DOCKER_APT_PACKAGES` remains a legacy fallback and is documented as such.

Real environment tested: local macOS checkout plus Blacksmith Testbox Linux runner through Crabbox.

Exact steps or command run after this patch:

```sh
pnpm docs:list
node scripts/run-vitest.mjs run --config test/vitest/vitest.e2e.config.ts src/docker-setup.e2e.test.ts
node scripts/run-vitest.mjs src/dockerfile.test.ts test/scripts/test-install-sh-docker.test.ts
pnpm exec oxfmt --check --threads=1 docs/install/docker.md docs/install/podman.md scripts/clawdock/README.md docs/help/faq.md CHANGELOG.md
git diff --check origin/main...HEAD
.agents/skills/autoreview/scripts/autoreview --mode local
.agents/skills/autoreview/scripts/autoreview --mode branch
pnpm check:changed
```

Evidence after fix:

- `src/docker-setup.e2e.test.ts`: 34 tests passed with the e2e Vitest config.
- `src/dockerfile.test.ts` + `test/scripts/test-install-sh-docker.test.ts`: 32 tests passed.
- `oxfmt --check`: passed.
- `git diff --check origin/main...HEAD`: passed.
- `autoreview --mode local`: clean.
- `autoreview --mode branch`: clean.
- `pnpm check:changed`: passed in Blacksmith Testbox `tbx_01krwqmfhcdekaczvrkxnb7t59`, GitHub Actions run https://github.com/openclaw/openclaw/actions/runs/26014630478, exit code 0.
- PR CI for head `50469900ec91a6bebb9184848a50211d8d968378`: green or still pending at comment time; Real behavior proof passed after maintainer override.

Observed result after fix: setup tests cover new-var precedence, legacy fallback, explicit empty new var suppressing legacy, and Dockerfile ARG handling without a baked default. Docs now use plugin wording and describe the legacy fallback consistently.

What was not tested: full Docker image build with a local Docker daemon on this host; the local daemon was unavailable. Contributor supplied live Podman build output in the PR body, and maintainer proof covered tests, formatting, docs, branch review, and Testbox changed checks.
